File: py-codellama13B-FP-all/taskid-6_parse_nested_parens-gen6.py
from typing import List

def parse_nested_parens(paren_string: str) -> List[int]:
    """ Input to this function is a string represented multiple groups for nested parentheses separated by spaces.
    For each of the group, output the deepest level of nesting of parentheses.
    E.g. (()()) has maximum two levels of nesting while ((())) has three.

    >>> parse_nested_parens('(()()) ((())) () ((())()())')
    [2, 3, 1, 3]
    """

    def _count_nested_parens(string: str) -> int:
        level = 0
        for ch in string:
            if ch == '(':
                level += 1
            elif ch == ')':
                level -= 1
        return level

    return [_count_nested_parens(string.strip()) for string in paren_string.split()]

File: java-codellama7B-AWQ-all/taskid-6_parse_nested_parens-gen6.java
import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    // Input to this function is a string represented multiple groups for nested parentheses separated by spaces.
    // For each of the group, output the deepest level of nesting of parentheses.
    // E.g. (()()) has maximum two levels of nesting while ((())) has three.
    // >>> parseNestedParens(("(()()) ((())) () ((())()())"))
    // (new ArrayList<Long>(Arrays.asList((long)2l, (long)3l, (long)1l, (long)3l)))
    public static ArrayList<Long> parseNestedParens(String paren_string) {

        // Your code here
        ArrayList<Long> output = new ArrayList<Long>();
        for (String paren_group : paren_string.split(" ")) {
            int nest_level = 0;
            int level_len = 0;
            for (int i = 0; i < paren_group.length(); i++) {
                if (paren_group.charAt(i) == '(') {
                    nest_level++;
                }
                if (paren_group.charAt(i) == ')') {
                    nest_level--;
                    level_len += nest_level;
                }
            }
            output.add(level_len);
        }
        return output;
    }
    public static void main(String[] args) {
    assert(parseNestedParens(("(()()) ((())) () ((())()())")).equals((new ArrayList<Long>(Arrays.asList((long)2l, (long)3l, (long)1l, (long)3l)))));
    assert(parseNestedParens(("() (()) ((())) (((())))")).equals((new ArrayList<Long>(Arrays.asList((long)1l, (long)2l, (long)3l, (long)4l)))));
    assert(parseNestedParens(("(()(())((())))")).equals((new ArrayList<Long>(Arrays.asList((long)4l)))));
    }

}
